app/services/project_data_service.rb

```diff
@@ -52,8 +52,26 @@ def update_dependencies
   end
 
   def update_dependency(name, version)
+    update_required = dependency_update_required(name, version)
     @project.dependencies.find_or_initialize_by(name: name)
-            .update!(version: version)
+            .update!(version: version, update_required: update_required)
+  end
+
+  def dependency_update_required(name, version)
+    expectation = Horizon.config.stringify_keys["expected_version_#{name}"].split('.')
```

Comment:
This seems like it would fail upon empty config. This is fine if you end up going with my suggestion to implement defaults directly in `_config.rb`, but otherwise should have a guard.

(Minor: by convention this method's name should also end in `?`.)